### ISSUE TITLE
dataset in hf

### ISSUE THREAD

**Issue by LukeLIN-web** [CIntellifusion/VideoDPO#14]
https://huggingface.co/datasets/chungimungi/VideoDPO-10k  spend 30mins to find it

**Comment by LukeLIN-web** [comment]
● 训练在第一个 step 后崩溃了。错误是 DDP 与 gradient checkpointing 不兼容："Expected to mark a variable ready only once"。这是因为 DPO loss 对同一模型参数做了多次 forward-backward（训练模型
  + ref模型），加上 use_checkpoint: true，导致 DDP 参数标记冲突。

  需要在 DDP strategy 中设置 static_graph=True。让我找到配置位置。

● Explore(Find DDP strategy config)
  ⎿  Done (9 tool uses · 30.5k tokens · 28s)
  (ctrl+o to expand)

● 已找到问题所在。train_utils.py 里已有 find_unused_parameters: True，但缺少 static_graph: True。DPO loss 中 gradient checkpointing 和 DDP 配合需要 static graph。
                                                                                                                                                   

● Update(scripts/train_utils.py)
  ⎿  Added 1 line
      154          "target": "pytorch_lightning.strategies.DDPStrategy",
      155          "params": {
      156              "find_unused_parameters": True,
      157 +            "static_graph": True,                                                                                                                                           
      158          }
      159      }
      160      if "strategy" in lightning_config:

**Comment by LukeLIN-web** [comment]
lvdm/modules/utils.py:122 明确用了 use_reentrant=True，这就是和 DDP 冲突的根源。改成 False：
